cpp/src/io/orc/stripe_enc.cu

```diff
@@ -402,16 +402,11 @@ static __device__ uint32_t IntegerRLE(
       } else {
         intrle_minmax(vmax, vmin);
       }
-      vmin = min(vmin, (T)shuffle_xor(vmin, 1));
-      vmin = min(vmin, (T)shuffle_xor(vmin, 2));
-      vmin = min(vmin, (T)shuffle_xor(vmin, 4));
-      vmin = min(vmin, (T)shuffle_xor(vmin, 8));
-      vmin = min(vmin, (T)shuffle_xor(vmin, 16));
-      vmax = max(vmax, (T)shuffle_xor(vmax, 1));
-      vmax = max(vmax, (T)shuffle_xor(vmax, 2));
-      vmax = max(vmax, (T)shuffle_xor(vmax, 4));
-      vmax = max(vmax, (T)shuffle_xor(vmax, 8));
-      vmax = max(vmax, (T)shuffle_xor(vmax, 16));
+#pragma unroll 5
+      for (int i = 1; i <= 16; i *= 2) {
```

Comment:
This could be replaced by a `transform_reduce` with a `minmax` reduction operation.

Reply:
Better yet, is this just a warp-level min&max reduction? Just use a CUB warp reduce.